aiida/engine/processes/calcjobs/calcjob.py

```diff
@@ -97,18 +97,39 @@ def validate_calc_job(inputs: Any, ctx: PortNamespace) -> Optional[str]:  # pyli
     return None
 
 
+def validate_stash_options(stash_options: Any, _: Any) -> Optional[str]:
+    """Validate the ``stash`` options."""
+    from aiida.common.datastructures import StashMode
+
+    target_base = stash_options.get('target_base', None)
+    source_list = stash_options.get('source_list', None)
+    stash_mode = stash_options.get('mode', StashMode.COPY.value)
+
+    if not isinstance(target_base, str) or not os.path.isabs(target_base):
+        return f'`metadata.options.stash.target_base` should be an absolute filepath, got: {target_base}'
+
+    if not isinstance(source_list, list) or any(not isinstance(src, str) or os.path.isabs(src) for src in source_list):
+        return f'`metadata.options.stash.source_list` should be a list of relative filepaths, got: {source_list}'
```

Comment:
I'm confused with the specification in the calcjob. There it says `source_list` could be tuple or list, but here it seems to only accept lists. Shouldn't it be list or tuple or is there some subclassing relationship I'm not aware of? In any case, wouldn't this already be checked automatically?

Reply:
I have added tuple as a valid type. It is true that the `valid_type` in the port declaration should already catch this, but I still include it here to make the validator more robust. If somebody changes the `valid_type` the conditional in the validator will not fail in the second part when it tries to iterate over the value when it is not fact iterable.

Reply:
note, I think these days, rather than using `isinstance(obj, (list, tuple))` its better to use `isinstance(obj, Sequence)`. But not a deal breaker 😉

Reply:
I used to do this, but I have bitten by this before. The `Sequence` does not only match `list` and `tuple` and sometimes the other matches are not anticipated and can cause problems. So I prefer to be explicit now

Reply:
fair